lib/srv/discovery/common/interfaces.go

```diff
@@ -26,8 +26,12 @@ import (
 type Fetcher interface {
 	// Get returns the list of resources from the cloud after applying the filters.
 	Get(ctx context.Context) (types.ResourcesWithLabels, error)
+	// MatchingLabels returns the labels that the fetcher is matching.
+	MatchingLabels() types.Labels
 	// ResourceType identifies the resource type the fetcher is returning.
 	ResourceType() string
+	// ResourceSubType identifies the resource sub-type the fetcher is returning.
+	MatchesResource(types.ResourceWithLabels) bool
```

Comment:
nit: update doc.

do you think `MatchesResource` can do label matching internally?